src/aws-cpp-sdk-core/include/aws/core/http/HttpTypes.h

```diff
@@ -20,43 +20,37 @@ namespace Aws
         /**
          * Models Http methods.
          */
-        enum class HttpMethod
-        {
-            HTTP_GET,
-            HTTP_POST,
-            HTTP_DELETE,
-            HTTP_PUT,
-            HTTP_HEAD,
-            HTTP_PATCH
-        };
-
-        /**
-         * Possible default http factory vended http client implementations.
-         */
-        enum class TransferLibType
-        {
-            DEFAULT_CLIENT = 0,
-            CURL_CLIENT,
-            WIN_INET_CLIENT,
-            WIN_HTTP_CLIENT
-        };
-
-        /**
-         * Configuration for a HTTP client, currently used only by libCurl
-         */
-        enum class TransferLibPerformanceMode
-        {
-            LOW_LATENCY = 0, // run http client for a lower latency, at the expense of CPU
-            REGULAR
-        };
-
-        namespace HttpMethodMapper
-        {
-            /**
-             * Gets the string value of an httpMethod.
-             */
-            AWS_CORE_API const char* GetNameForHttpMethod(HttpMethod httpMethod);
-        } // namespace HttpMethodMapper
+    enum class HttpMethod {
```

Comment:
I still don't understand why HttpTypes.h file is in this PR and how it is related to the Smithy Refactor.
There are only formatting change and new enum values added
>      HTTP_CONNECT,
>      HTTP_OPTIONS,
>      HTTP_TRACE,

but there are no context why they are added here.